HSND-383 Refactor/handler types (#10)
Homeservenow/serverless-http-handler: readme.md

 ## Usage

 ```typescript
import {httpHandler, NotFoundException} from '@homeservenow/serverless-aws-handler';
import {APIGatewayEvent} from 'aws-lambda';
import {database} from './database';
import { httpHandler, NotFoundException } from '@homeservenow/serverless-aws-handler';
import { database } from './database';

> export const getCatHandler = httpHandler(async (event: APIGatewayEvent): Promise<CatInterface> => {

Author: Let's do a pass on the readme after merging this :)

`UnProcessableEntityException` | 422 | 'Unprocessable Entity
`InternalServerError` | 500 | Internal Server Error

## Validator

```typescript
export const myHandler = httpHandler({
    handler: myHandlerMethod,
});
```

## Serialisation

#### Input

#### Output

## Custom Error Handling

## Default status code and headers

The below example will always output a status code of 204.

```typescript
export const myHandler = httpHandler({
    handler: (event: APIGatwayProxyEvent) => {

    },
    defaultStatusCode: HttpStatusCode.NO_CONTENT,
});
```
#### default Headers 

> The below example will always a header `X-header`

Author: ```suggestion
The below example will always output a header `X-header`
```

```typescript
export const myHandler = httpHandler({
    handler: (event: APIGatwayProxyEvent) => {

    },
    defaultHeaders: {
        ['X-header']: 'Hello!',
    },
});
```

## Exception logging options 

> The httpHandler has it's own logging method which you can customise. This method will make info logs, warn logs and error logs using console.

Author: ```suggestion
The httpHandler has its own default logging method which you can customise. This method will make info logs, warn logs and error logs using console.
```

There are several different logging methods 

- By http status
- Range by http status
- Blacklist | whitelist
- boolean

> Logging is only called when an exception occurs. All of the below examples are for generally for status codes of 400 and above. However it is possible to override.

Author: ```suggestion
Logging is only called when an exception occurs. All of the below examples are generally for status codes of 400 and above. However it is possible to override.
```

#### Singular http status

Will only log `INTERNAL_SERVER_ERROR` (500) status codes.

```typescript
export const myHandler = httpHandler({
    handler: myHandlerMethod,
    loggingHandlingOpions: HttpStatusCode.INTERNAL_SERVER_ERROR,
});
```

#### Range by http status

> A range will only log errors within the targed range including the given values. In this example all exceptions with a status code between `400` and `403` will be logged. `500` Status codes will not be logged in this example.

Author: ```suggestion
A range will only log errors within the target range including the given values. In this example all exceptions with a status code between `400` and `403` will be logged. `500` Status codes will not be logged in this example.
```

```typescript
export const myHandler = httpHandler({
    handler: myHandlerMethod,

>     loggingHandlingOpions: [HttpStatusCode.BAD_REQUEST, HttpStatusCode.FORBIDDEN],

Author: ```suggestion
    loggingOptions: [HttpStatusCode.BAD_REQUEST, HttpStatusCode.FORBIDDEN],
```
